### ISSUE TITLE
Add support for specifying schema via `# $schema: …`

### ISSUE THREAD

**Issue by pascal-hofmann** [redhat-developer/yaml-language-server#959]
### Is your enhancement related to a problem? Please describe.
JetBrains recently added support for specifying YAML schema via comment to their products. This looks similar to the modeline that can be used for yaml-language-server.

```
# $schema: …
```

Examples:
```
# $schema: https://example.com/path/to/schema.json
# $schema: ./schema.json
```

### Describe the solution you would like
It would be great if yaml-language-server would support both ways of specifying the schema inline, so we don't have to specify it twice.

### Describe alternatives you have considered
None.

**Comment by jeremyfiel** [comment]
duplicate #959

**Comment by pascal-hofmann** [comment]
> duplicate #959

This is #959.

**Comment by cromefire** [comment]
Currently it's not quite a duplicate of #950 as they made a crucial mustake that actually makes it incompatible with JetBrains' IDEs (`=` instead of `:`).

**Comment by tysonclugg** [comment]
> duplicate #950

No, #950 is about getting the schema from inside the data strucutre, *without using a modeline*:
```yaml
$schema: ./schema.json
foo: "1"
```

This issue is about adjusting the modeline parser to allow more relaxed forms of the schema declaration (ie: without the `yaml-language-server` guff) *when using a modeline* (aka: special comment):
```yaml
# $schema=./schema.json
foo: "1"
```

**Comment by pietdevries94** [comment]
It seems this is a works nowadays. Let's close this issue :)

**Comment by datho7561** [comment]
Fixed in #1193
